.github/workflows/docker-build.yml

```diff
@@ -15,9 +15,10 @@ jobs:
         id: set-tag
         run: |
           SHORT_SHA=$(git rev-parse --short HEAD)
-          if [[ "${{ github.ref }}" == "refs/heads/main" ]]; then
+          TAG_VERSION=$(echo ${{ github.ref }} | sed 's,refs/tags/v,,g')
+          if [[ "${{ github.ref }}" == "refs/tags/v*" ]]; then
```

Comment:
Please also change this check to use the ref_name and check whether it's possible to use the "official" semver regex for the check: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string

Or use this github action to check it the tag is a valid semver: https://github.com/marketplace/actions/check-version-format-in-tag

If we use this check, the tag `velociraptor` is a valid tag and will trigger a release, with the latest tag as well. Something like that isn't likely to happen, but we should still due our due diligence here and build the check properly.

Reply:
github.ref is also chosen carefully here, as the workflow runs on every push as intended.
 
An actor could name a branch v1.2.3 to trigger these actions unexpectedly since github.ref_name would just check the name without distinguishing between a branch and a tag.

- Example: `if [[ "${{ github.ref_name }}" == v* ]]` might be true for both a tag v1.2.3 and a branch named v1.2.3.

Testing the version of the tag according to semver specifications seems to me to be a valid point

Reply:
Yeah, valid point. As mentioned on my other response, the github context also provides a ref_type to differentiate between branches and tags.

I still think we should separate the dev builds from the tagged release ones, as we do for example in the sparrow repo.

Reply:
- https://github.com/caas-team/py-kube-downscaler/pull/21/commits/0a8c46e07f6f9d722c685efad1e670767c3fb908